llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp

```diff
@@ -329,6 +374,15 @@ bool PointerReplacer::collectUsersRecursive(Instruction &I) {
         ValuesToRevisit.insert(Inst);
         continue;
       }
+
+      auto *TruePtr = dyn_cast<PointerType>(SI->getTrueValue()->getType());
+      auto *FalsePtr = dyn_cast<PointerType>(SI->getFalseValue()->getType());
```

Comment:
Select operands must be the same type. There is no situation where these can be different

Reply:
I see... will need to rework it some to detect it properly. Thanks.